### PR TITLE
refactor(model): update hydro forms

### PR DESCRIPTION
**Pull request by hdinia**
#1157

### CODE REVIEW

**Review by pl-buiquang — APPROVED** [review]

File: webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Hydro/utils.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import { v4 as uuid } from "uuid";
+import { MatrixStats } from "../../../../../../../common/types";
+
+////////////////////////////////////////////////////////////////
+// Enums
+////////////////////////////////////////////////////////////////
+
+export enum MatrixType {
+  Dailypower = "dailypower",
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Hydro/utils.ts]: String not necessary

File: webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Hydro/utils.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+////////////////////////////////////////////////////////////////
+
+export interface HydroMatrixProps {
+  title: string;
+  url: string;
+  cols?: string[];
+  stats?: MatrixStats;
+}
+
+type Matrices = Record<MatrixType | string, HydroMatrixProps>;
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Hydro/utils.ts]: Remove ` | string`

File: webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Hydro/utils.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  type: string;
+}
+
+////////////////////////////////////////////////////////////////
+// Constants
+////////////////////////////////////////////////////////////////
+
+export const HYDRO_ROUTES: HydroRoute[] = [
+  {
+    id: uuid(),
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Hydro/utils.ts]: id not necessary, use `path`

**Review by skamril — CHANGES_REQUESTED** [review]

File: webapp/src/components/App/index.tsx [hunk 4]
```diff
@@ -85,37 +82,13 @@ function App() {
...
-                              <Route
-                                path="watervalues"
-                                element={<WaterValues />}
-                              />
-                              <Route
-                                path="timeseries"
-                                element={<Navigate to="hydrostorage" replace />}
-                              />
-                              <Route path="timeseries" element={<TimeSeries />}>
+                              {HYDRO_ROUTES.map((route: HydroRoute) => (
```
Comment (skamril) [inline webapp/src/components/App/index.tsx:85]: Tips: `R.map` keeps the type

**Review by skamril — CHANGES_REQUESTED** [review]

File: webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Hydro/utils.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+////////////////////////////////////////////////////////////////
+// Types
+////////////////////////////////////////////////////////////////
+
+export interface HydroMatrixProps {
+  title: string;
+  url: string;
+  cols?: string[];
+  stats?: MatrixStats;
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Hydro/utils.ts]: `stats` is mandatory

File: webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Hydro/utils.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+  WaterValues,
+  HydroStorage,
+  RunOfRiver,
+}
+
+////////////////////////////////////////////////////////////////
+// Types
+////////////////////////////////////////////////////////////////
+
+export interface HydroMatrixProps {
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Hydro/utils.ts]: Remove export

File: webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Hydro/HydroMatrix.tsx [hunk 7]
```diff
@@ -19,14 +23,14 @@ function ReservoirLevels() {
...
       <MatrixInput
-        title="Reservoir levels"
-        columnsNames={reservoirLevelsCols}
+        title={hydroMatrix.title}
+        columnsNames={hydroMatrix.cols}
         study={study}
-        url={reservoirLevelsUrl}
-        computStats={MatrixStats.TOTAL}
+        url={hydroMatrix.url.replace("{areaId}", areaId)}
+        computStats={hydroMatrix.stats || MatrixStats.NOCOL}
```
Comment (skamril) [inline webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Hydro/HydroMatrix.tsx]: Remove `|| MatrixStats.NOCOL`

**Review by skamril — CHANGES_REQUESTED** [review]

**Review by skamril — APPROVED** [review]

**Pull request merged**